x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -0,0 +1,374 @@
+An optional regular expression can be configured via the realm setting `claims.principal_pattern` to extract a substring.
+
+* The _groups_ claim from the JWT is optional. It is configurable via the realm setting `claims.groups`.
+An optional regular expression can be configured via the realm setting `claims.groups_pattern` to extract a substring value.
+
+* The _name_ claim from the JWT is optional. It is configurable via the realm setting `claims.name`.
+An optional regular expression can be configured via the realm setting `claims.name_pattern` to extract a substring value.
+
+* The _mail_ claim from the JWT is optional. It is configurable via the realm setting `claims.mail`.
+An optional regular expression can be configured via the realm setting `claims.mail_pattern` to extract a substring value.
+
+* The _dn_ claim from the JWT is optional. It is configurable via the realm setting `claims.dn`.
+An optional regular expression can be configured via the realm setting `claims.dn_pattern` to extract a substring value.
+
+[[jwt-authorization]]
+==== JWT realm authorization
+
+JWT realm authorization supports role mapping API or delegated authorization realms.
+
+WARNING: JWT realm does not support using the `role_mapping.yml` file.
+
+===== JWT realm authorization via Role Mapping API
+
+.If JWT realm role mapping is used for authorization, these claims are available for role mapping.
+* `principal` claim: JWT claim in the form of a string is used as the {es} user's principal string value.
+* `dn` claim: JWT claim in the form of a Distinguished Name string is used as the {es} user's DN string value.
+* `groups` claim: JWT claim in the form of a comma separate list is used as the {es} user's groups list value.
+* `metadata` claims: JWTs claims in the form of strings, numbers, booleans, and collections are used as the {es} user's metadata, formatted as `metadata.jwt_claim_<key>` = `<value>` pairs.
+
+See <<ref-jwt-settings>>.
+
+===== Example JWT Realm Role Mapping Rule
+
+Example JWT realm role mapping rule.
+
+[source,console]
+----
+PUT /_security/role_mapping/jwt1_users
+{
+  "roles" : [ "user" ],
+  "rules" : { "all" : [
+      { "field": { "realm.name": "jwt1" } },
+      { "field": { "username": "principalname1" } }
+      { "field": { "dn": "CN=Principal Name 1,DC=example.com" } }
+      { "field": { "groups": "group1" } }
+      { "field": { "metadata.jwt_claim_other": "other1" } }
+  ] },
+  "enabled": true
+}
+----
+
+===== JWT realm authorization via authorization delegation
+
+.If Delegated Realms are used for authorization, this claim is available for roles lookup.
+* `principal` claim: JWT claim in the form of a string is used as the {es} user's principal value.
+
+See <<ref-jwt-settings>>.
+
+WARNING: When Delegated Authorization is enabled in JWT realm, claims for `groups`, `dn`, and `metadata` not used for the {es} user's values.
```

Comment:
```suggestion
WARNING: When Delegated Authorization is enabled in JWT realm, claims for `groups`, `dn`, `mail`, `name`, and `metadata` not used for the {es} user's values.
```

I think I missed mentioning `mail` and `name` here when I wrote the first draft.